src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java

```diff
@@ -0,0 +1,400 @@
+     * The exercise settings override the course settings, but depending on the sub settings type, the combining strategy is different.
+     * ChatSettings: exercise settings are mandatory for the chat feature to be enabled
+     * HestiaSettings: exercise settings are optional for the hestia feature to be enabled
+     *
+     * @param programmingExercise the programming exercise for which to get the settings
+     * @param reduced             if true only the enabled flag is combined, otherwise all settings are combined
+     * @return the combined IrisSettings
+     */
+    public IrisSettings getCombinedIrisSettings(ProgrammingExercise programmingExercise, boolean reduced) {
+        var courseSettings = getCombinedIrisSettings(programmingExercise.getCourseViaExerciseGroupOrCourseMember(), reduced);
+        var exerciseSettings = getIrisSettingsOrDefault(programmingExercise);
+
+        var combinedSettings = new IrisSettings();
+        combinedSettings.setIrisChatSettings(combineSubSettings(courseSettings.getIrisChatSettings(), exerciseSettings.getIrisChatSettings(), false, reduced));
+        combinedSettings.setIrisHestiaSettings(combineSubSettings(courseSettings.getIrisHestiaSettings(), exerciseSettings.getIrisHestiaSettings(), true, reduced));
+        return combinedSettings;
+    }
+
+    /**
+     * Combines the course and exercise sub-settings together.
+     * The exercise settings override the course settings, but depending on the exerciseSettingsAreOptional parameter,
+     * the combining strategy is different. If exerciseSettingsAreOptional is true, the course settings are used in full
+     * if the exercise settings are null. Otherwise the exercise settings have to be present.
+     *
+     * @param subSettings1                 The course settings
+     * @param subSettings2                 The exercise settings
+     * @param secondSubSettingsAreOptional Whether the exercise settings are optional or not
+     * @param reduced                      Whether only the enabled flag should be combined or all settings
+     * @return The combined sub-settings
+     */
+    private IrisSubSettings combineSubSettings(IrisSubSettings subSettings1, IrisSubSettings subSettings2, boolean secondSubSettingsAreOptional, boolean reduced) {
+        if (secondSubSettingsAreOptional && subSettings2 == null) {
+            return subSettings1;
+        }
+
+        var combinedSettings = new IrisSubSettings();
+
+        var enabled = subSettings2 != null && subSettings2.isEnabled() && subSettings1 != null && subSettings1.isEnabled()
+                && applicationContext.getEnvironment().acceptsProfiles(Profiles.of("iris-gpt3_5"));
+        combinedSettings.setEnabled(enabled);
+
+        if (!reduced) {
+            var preferredModel = subSettings2 != null && subSettings2.getPreferredModel() != null ? subSettings2.getPreferredModel()
```

Comment:
Ternary is hard to read here. Maybe consider something like this:
```suggestion
            var preferredModel = null;
            if (subSettings2 != null && subSettings2.getPreferredModel() != null) {
                preferredModel = subSettings2.getPreferredModel();
            } else if (subSettings1 != null) {
                preferredModel = subSettings1.getPreferredModel();
            }
```

Reply:
Fixed it in #6714.